src/ray/common/cgroup/README.md

```diff
@@ -0,0 +1,35 @@
+### Ray core cgroup documentation
+
+#### Physical execution mode
+
+Ray core supports a physical execution mode, which allows users to cap resource consumption for their applications.
+
+A few benefits:
+- It prevents application from eating up unlimited resource to starve other applications running on the same node;
+- It protects other processes on the node (including ray system components) to be killed due to insufficient resource (i.e. OOM);
+
+TODO(hjiang): reserve minimum resource will be supported in the future.
+
+#### Disclaimer and presumption
+
+- The feature is built upon cgroup, which only supports linux;
+- Only cgroup v2 is supported, meanwhile ray also requires application to have write permission and cgroup v2 be mounted in rw mode;
+- At the initial version, ray caps max resource usage via heuristric estimation (TODO: support user passed-in value).
+
+#### Implementation details
```

Comment:
[general] I am a bit worried about cgroup leaks on (1) subprocesses holding the cgroup can't be deleted on task exit, (2) raylet crash. Do you have some good ideas on this?

For (1) we certainly don't want these procs. We already use linux subreaper to kill leaked procs via [RAY_kill_child_processes_on_worker_exit_with_raylet_subreaper](https://github.com/ray-project/ray/blob/c79de07ba31d1561e121f96dd04e1bcbf3a3a92a/src/ray/common/ray_config_def.h#L906) and maybe we can have another flag like `RAY_kill_child_processes_on_worker_exit_with_cgroup` so on cgroup exit we SIGKILL all procs in cgroup. WDYT?

For (2) I can't think of a good idea. If you SIGKILL raylet, all cgroups are destined to leak. Maybe we can just accept it. But if so we need to document it both in here and in the eventual user facing document.

Also we have a Q (3): do we want to restrict processes created by raylet to don't have permission to move their own cgroups? Well, Ray does not have a lot of security features so we might be OK with this.

Reply:
> (1) subprocesses holding the cgroup can't be deleted on task exit

I see, integrating the cgroup cleanup is a good idea, but we might need a way to get cgroup folder name with unknown PID. Or a hacky way is to have another setup of cgroups which cannot be deleted after task / actor completion.

Reply:
> For (2) I can't think of a good idea. If you SIGKILL raylet, all cgroups are destined to leak. Maybe we can just accept it. But if so we need to document it both in here and in the eventual user facing document.

I cannot think of a good way either.

Reply:
> Also we have a Q (3): do we want to restrict processes created by raylet to don't have permission to move their own cgroups? Well, Ray does not have a lot of security features so we might be OK with this.

Don't think it's something we could handle at the moment; theoretically we are not container-ize user process, so it could access everything in the node / filesystem.

Reply:
> [general] I am a bit worried about cgroup leaks on (1) subprocesses holding the cgroup can't be deleted on task exit

At the initial version, I would like to implement a simple RAII-style cgroup setup and cleanup for raylet.